Assets/Scripts/Models/Character.cs

```diff
@@ -1017,9 +999,9 @@ private bool CheckForJobMaterials()
 
     private bool WalkingToUsableInventory()
     {
-        bool destHasInventory = pathAStar != null && pathAStar.EndTile() != null && pathAStar.EndTile().Inventory != null;
+        bool destHasInventory = movementPath != null && movementPath.Last() != null && movementPath.Last().Inventory != null;
```

Comment:
I've changed the code to avoid the null exception, using @sboigelot's suggestion. 

As for booking material, I think that's a whole other PR.